src/scripts/main.js

```diff
@@ -1 +1,90 @@
 'use strict';
+
+// Select the notification div
+const notificationDiv = document.querySelector('[data-qa="notification"]');
+
+// Helper function to display messages
+function showNotification(message, isError = false) {
+  notificationDiv.textContent = message;
+  notificationDiv.className = isError ? 'error' : 'success';
+  notificationDiv.style.display = 'block';
+
+  // Hide the notification after 3 seconds
+  setTimeout(() => {
+    notificationDiv.style.display = 'none';
+  }, 3000);
+}
+
+// First Promise
+const firstPromise = new Promise((resolve, reject) => {
+  let isHandled = false; // Flag to ensure only one handler is executed
+
+  const clickHandler = () => {
+    if (!isHandled) {
+      isHandled = true;
+      resolve('First promise was resolved');
+      document.removeEventListener('click', clickHandler);
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  setTimeout(() => {
+    if (!isHandled) {
+      isHandled = true;
+      reject(new Error('First promise was rejected'));
+      document.removeEventListener('click', clickHandler);
+    }
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => showNotification(message))
+  .catch((error) => showNotification(error.message, true));
+
+// Second Promise
+const secondPromise = new Promise((resolve) => {
+  const clickHandler = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      // Left or Right click
+      resolve('Second promise was resolved');
+      document.removeEventListener('click', clickHandler);
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+});
```

Comment:
The second promise does not have a timeout or rejection case, which means it will remain unresolved if no click occurs. Consider adding a timeout with a rejection to handle cases where no click happens.